Kickstarter-iOS/Views/Controllers/PledgeTableViewController.swift

```diff
@@ -51,10 +52,30 @@ class PledgeTableViewController: UITableViewController {
 
     self.viewModel.outputs.reloadWithData
       .observeForUI()
-      .observeValues { [weak self] (amount, currency, delivery, isLoggedIn) in
-        self?.dataSource.load(amount: amount, currency: currency, delivery: delivery, isLoggedIn: isLoggedIn)
+      .observeValues { [weak self] data in
+        self?.dataSource.load(data: data)
 
         self?.tableView.reloadData()
+
+        self?.viewModel.inputs.reloadData()
+    }
+
+    self.viewModel.outputs.selectedShippingRuleData
+      .observeForUI()
+      .observeValues { [weak self] selectedShippingRuleData in
+        self?.dataSource.loadSelectedShippingRule(data: selectedShippingRuleData)
+
+        guard let shippingIndexPath = self?.dataSource.shippingCellIndexPath() else {
```

Comment:
Perhaps a helper function on the dataSource like `containsShippingCell() -> Bool` which you could also filter on to avoid the `guard`

```swift
.observeForUI()
.filter { [weak self] _ in self?.dataSource.containsShippingCell() == .some(false) }
```

Reply:
`IndexPath` still needs to be optional as a return type from `self?.dataSource.shippingCellIndexPath()`, so we'll still need to unwrap it somehow even if we filter the signal against it not containing the shipping cell. I don't think the filter really does much in that case 🤔